src/gtk/control.cpp

```diff
@@ -350,9 +350,14 @@ wxSize wxControl::GTKGetPreferredSize(GtkWidget* widget) const
     if ( wasHidden )
         gtk_widget_show(widget);
 
-    gtk_widget_set_size_request(widget, -1, -1);
+    // Temporarily unset the requested size if it's set.
+    if ( w != -1 || h != -1 )
```

Comment:
I don't think there is anything to be gained by adding these checks, `gtk_widget_set_size_request()` won't do anything if the values aren't being changed.

Reply:
AFAICS this ends up calling `g_object_freeze_notify()` and `g_object_thaw_notify()` even if nothing changes and I'm not sure if these calls are really free. In any case, for me this is more philosophical than practical: why bother calling a function if we know it won't do anything?

Reply:
> I'm not sure if these calls are really free

While they won't literally do nothing, they will have no effect and their cost is negligible.

> why bother calling a function if we know it won't do anything?

Surely the "bother" is code to avoid the call. I see it as the equivalent of checking for a divisor of one before performing a division. Sure, you can do it, but why bother?

Reply:
OK, I'll drop this commit.